src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -491,13 +492,32 @@ public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, HttpCompl
             CancellationTokenSource cts;
             bool disposeCts;
             bool hasTimeout = _timeout != s_infiniteTimeout;
+            TimeoutState timeoutState = null; // Lazy initialization to avoid unnecessary alocation when timeout is not set
+            Timer timer = null;
             if (hasTimeout || cancellationToken.CanBeCanceled)
             {
                 disposeCts = true;
                 cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, _pendingRequestsCts.Token);
                 if (hasTimeout)
                 {
-                    cts.CancelAfter(_timeout);
+                    long timeoutMilliseconds = (long)_timeout.TotalMilliseconds;
+                    if (timeoutMilliseconds < -1 || timeoutMilliseconds > int.MaxValue)
```

Comment:
The timeout value validity is checked in ```Timeout``` setter, it's not necessary here per se.

Reply:
Yes, you are right, I overlooked this. Moreover, conversion to long isn't needed at all since Timer ctor has TimeSpan overload. Code removed.